src/main/java/org/myrobotlab/service/Runtime.java

```diff
@@ -302,19 +311,114 @@ static public String getPublicGateway() {
+    return create(null, name, type);
+  }
+
+  static public synchronized ServiceInterface create(String name) {
+    return create(null, name, null);
+  }
+
+  /**
+   * Create create(name, type) goes through the full service lifecycle of:
+   * 
+   * <pre>
+   * clear - clearing the plan for construction of service(s) needed 
+   * load  - loading the plan for desired services 
+   * check - checking all planned service have met appropriate licensing and dependency checks create -
+   * </pre>
+   * 
+   * @param name
+   * @param type
+   * @return
+   */
+  static public synchronized ServiceInterface create(Plan plan, String name, String type) {
+    ServiceInterface si = Runtime.getService(name);
+    if (si != null) {
+      return si;
+    }
+
+    // Runtime.clear(); - turns out you don't want to do this
+    Runtime.load(plan, name, type); // FIXME evaluate appropriate return for
+                                    // load
+    Runtime.check(name, type);
+    // at this point - the plan should be loaded, now its time to create the
+    // children peers
+    // and parent service
+    return createServices(name);
+  }
+
+  /**
+   * creates all services necessary for this service - "all peers" and the
+   * parent service too
+   * 
+   * @param name
+   * @return
+   */
+  private static ServiceInterface createServices(String name) {
     Runtime runtime = Runtime.getInstance();
-    String filename = runtime.getConfigDir() + fs + runtime.getConfigName() + fs + name + ".yml";
-    File check = new File(filename);
-    if (!check.exists()) {
-      runtime.error("create(%s) called - but required file %s does not exist", name, filename);
+    // get plan
+    Plan plan = runtime.plan;
+
+    ServiceInterface ret = null;
+
+    // FIXME - Design Problem ... doesn't it make sense to "start" peers but not
+    // start the parent
+    // service ? Seems like the only reason to create without starting is a
+    // design problem
+    // of the implementation of the service (like webgui and it's port or
+    // autoStartBrowser) and
+    // the ability to load
+
+    // if Runtime definition exists - pre-load it
+    // special handling of runtime - runtime has a registry
+    // it will be loaded now into the plan so our plan will not be modified
+    // while executing it (bad idea)
+    RuntimeConfig rc = (RuntimeConfig) plan.get("runtime");
+    if (rc != null) {
+      for (String n : rc.registry) {
+        if (!"runtime".equals(n)) {
+          load(n);
+        }
+      }
+    }
+
+    // iterate through plan
+    for (String peerName : plan.keySet()) {
+      if ("runtime".equals(peerName)) {
+        // already handled runtime
+        // apply the config
+        runtime.apply(plan.get("runtime"));
+        continue;
+      }
+
+      ServiceConfig sc = plan.get(peerName);
+      if (sc.autoStart) {
+        ServiceInterface si = createService(peerName, sc.type, null);

```

Comment:
what's the 3rd argument here?  why is it null?

Reply:
id of the mrl instance ... its "always" been there, since first tried to do distributed mrl